beacon-chain/db/filesystem/data_column.go

```diff
@@ -0,0 +1,1014 @@
+	"github.com/OffchainLabs/prysm/v6/async/event"
+	"github.com/OffchainLabs/prysm/v6/beacon-chain/verification"
+	fieldparams "github.com/OffchainLabs/prysm/v6/config/fieldparams"
+	"github.com/OffchainLabs/prysm/v6/config/params"
+	"github.com/OffchainLabs/prysm/v6/consensus-types/blocks"
+	"github.com/OffchainLabs/prysm/v6/consensus-types/primitives"
+	"github.com/OffchainLabs/prysm/v6/io/file"
+	"github.com/OffchainLabs/prysm/v6/time/slots"
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+)
+
+const (
+	version                            = 0x01
+	versionOffset                      = 0                           // bytes
+	versionSize                        = 1                           // bytes
+	maxSszEncodedDataColumnSidecarSize = 536_870_912                 // 2**(4*8) / 8 bytes
+	sidecarSizeOffset                  = versionOffset + versionSize // (Offset of the encoded size of the SSZ encoded data column sidecar)
+	sidecarSizeSize                    = 4                           // bytes (Size of the encoded size of the SSZ encoded data column sidecar)
+	mandatoryNumberOfColumns           = 128                         // 2**7
+	indicesOffset                      = sidecarSizeOffset + sidecarSizeSize
+	indicesSize                        = mandatoryNumberOfColumns
+	nonZeroOffset                      = mandatoryNumberOfColumns
+	headerSize                         = versionSize + sidecarSizeSize + mandatoryNumberOfColumns
+	dataColumnsFileExtension           = "sszs"
+	prunePeriod                        = 1 * time.Minute
+)
+
+var (
+	errWrongNumberOfColumns                 = errors.New("wrong number of data columns")
+	errDataColumnIndexTooLarge              = errors.New("data column index too large")
+	errWrongBytesWritten                    = errors.New("wrong number of bytes written")
+	errWrongVersion                         = errors.New("wrong version")
+	errWrongBytesHeaderRead                 = errors.New("wrong number of bytes header read")
+	errTooManyDataColumns                   = errors.New("too many data columns")
+	errWrongSszEncodedDataColumnSidecarSize = errors.New("wrong SSZ encoded data column sidecar size")
+	errDataColumnSidecarsFromDifferentSlots = errors.New("data column sidecars from different slots")
+)
+
+type (
+	// DataColumnStorage is the concrete implementation of the filesystem backend for saving and retrieving DataColumnSidecars.
+	DataColumnStorage struct {
+		base            string
+		retentionEpochs primitives.Epoch
+		fs              afero.Fs
+		cache           *dataColumnStorageSummaryCache
+		DataColumnFeed  *event.Feed
+		pruneMu         sync.RWMutex
+
+		mu      sync.Mutex // protects muChans
+		muChans map[[fieldparams.RootLength]byte]*muChan
+	}
+
+	// DataColumnStorageOption is a functional option for configuring a DataColumnStorage.
+	DataColumnStorageOption func(*DataColumnStorage) error
+
+	// DataColumnIdent is the unique identifier for a data column sidecar.
```

Comment:
```suggestion
	// DataColumnsIdent is the unique identifier for a data column sidecar.
```

Reply:
Fixed in affa5da5685b41d42bbbbfcf487fecd7619c2101, with a different wording.